javascript/src/main/pages/RoomSlots/RoomSlots.js

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect } from "react";
+import { useHistory } from "react-router-dom";
+import useSWR from "swr";
+import { Button } from "react-bootstrap";
+import { useAuth0 } from "@auth0/auth0-react";
+
+import { fetchWithToken } from "main/utils/fetch";
+import RoomSlotTable from "main/components/RoomSlots/RoomSlotTable";
+import Loading from "../../components/Loading/Loading";
+
+const RoomSlots = () => {
+  const {_user, getAccessTokenSilently: getToken } = useAuth0();
+  const { data: roleInfo } = useSWR(["/api/myRole", getToken], fetchWithToken);
+  const history = useHistory();
+
+  const { data: roomSlotList, error, mutate: mutateRoomSlots } = useSWR(
+    ["/api/public/roomslot", getToken],
+    fetchWithToken
+  );
+
+  useEffect(() => {
+    mutateRoomSlots();
+  }, [mutateRoomSlots]);
+
+  const isAdmin =
+    roleInfo && roleInfo.role && roleInfo.role.toLowerCase() === "admin";
```

Comment:
I believe this long series of checks can be simplified to something like:
```suggestion
  const isAdmin = roleInfo?.role?.toLowerCase() === "admin";
```